### PR TITLE
AOM-89: Test SingleAddon component

### PR DESCRIPTION
**Pull request by patlub**
### SUMMARY:

Adds unit tests for the SingleAddon component, which currently has no tests.

### CODE REVIEW

**Review by aenkya — COMMENTED** [review]
What is the coverage on this file?

**Comment by patlub** [comment]
70%

**Comment by aenkya** [comment]
Nice. LGTM

**Comment by joel-ace** [comment]
Looks Good

**Comment by Charpell** [comment]
Good job writing test for the SingleAddon component, however i feel testing for only the HTML tag element isn't concrete enough for the component.

**Comment by patlub** [comment]
@Charpell  Coverage file shows almost all lines covered. Besides component has 95% presentation logic.

**Comment by mkiterian** [comment]
As @Charpell has pointed out, where are tests for example that the expected app name, description or version is rendered? The mock has a name of `openmrs-addOnManager`. Shouldn't there be a test to verify that this is rendered?

**Comment by mkiterian** [comment]
How should the component behave if `app` is not provided? You should also add tests for the status icon and the action button.

File: tests/components/SingleAddon.test.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      "install":false,
+    }
+
+  const addonComponent =
+    mount(<SingleAddon
+            app={app}
+            addonParam={() => {}}
+            updatesVersion={{}} />);
+
+  it('Should render', () => {
```
Comment (mkiterian) [inline tests/components/SingleAddon.test.js]: Make the assertion messages more descriptive e.g `should render <SingleAddon  />`

File: tests/components/SingleAddon.test.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        "installs_allowed_from":null,
+        "default_locale":"en",
+        "deployed.owa.name":null,
+      }
+      ,"appType":"owa",
+      "install":false,
+    }
+
+  const addonComponent =
+    mount(<SingleAddon
```
Comment (mkiterian) [inline tests/components/SingleAddon.test.js]: Look into the usage of `shallow` vs `mount` and when one is preferred over the other

**Comment by mkiterian** [comment]
You have a mock for an OWA addon, how about a for a module? Is there a difference in the component for the two cases?

File: tests/components/SingleAddon.test.js (hunk 2 above)
Comment (patlub) [reply tests/components/SingleAddon.test.js]: okay

**Comment by patlub** [comment]
Cool. Let me improve my tests.

**Comment by Charpell** [comment]
It looks good now. 
Great Job :+1:

**Comment by dkayiwa** [comment]
@patlub have you addressed all the above review comments?

**Comment by patlub** [comment]
@dkayiwa Yes, I have addressed them all.

File: app/js/components/manageApps/ManageApps.jsx [hunk 3]
```diff
@@ -278,8 +278,10 @@ export default class ManageApps extends React.Component {
           uploadStatus: 0,
           showProgress: false,
           files: null,
+          updatesAvailable: null,
```
Comment (dkayiwa) [inline app/js/components/manageApps/ManageApps.jsx]: Could you explain how changes in this file are related to the test that the ticket requires?

File: app/js/components/manageApps/SingleAddon.jsx [hunk 4]
```diff
@@ -105,3 +106,10 @@ export default class SingleAddon extends React.Component{
     );
   }
 }
+
```
Comment (dkayiwa) [inline app/js/components/manageApps/SingleAddon.jsx:109]: Same question as above.

Comment (patlub) [reply app/js/components/manageApps/SingleAddon.jsx:109]: @dkayiwa The person who wrote the component I was testing didn't add required props validation, so I added it.

**Pull request merged**